skyportal/handlers/api/followup_request.py

```diff
@@ -1375,3 +1400,197 @@ async def put(self):
+              required: true
+              schema:
+                type: integer
+          responses:
+            200:
+              content:
+                application/json:
+                  schema: SingleDefaultFollowupRequest
+            400:
+              content:
+                application/json:
+                  schema: Error
+        multiple:
+          description: Retrieve all default follow-up requests
+          tags:
+            - filters
+          responses:
+            200:
+              content:
+                application/json:
+                  schema: ArrayOfDefaultFollowupRequests
+            400:
+              content:
+                application/json:
+                  schema: Error
+        """
+
+        with self.Session() as session:
+            if default_followup_request_id is not None:
+                default_followup_request = session.scalars(
+                    DefaultFollowupRequest.select(
+                        session.user_or_token,
+                        options=[joinedload(DefaultFollowupRequest.allocation)],
+                    ).where(DefaultFollowupRequest.id == default_followup_request_id)
+                ).first()
+                if default_followup_request is None:
+                    return self.error(
+                        f'Cannot find DefaultFollowupRequestRequest with ID {default_followup_request_id}'
+                    )
+                return self.success(data=default_followup_request)
+
+            default_followup_requests = (
+                session.scalars(
+                    DefaultFollowupRequest.select(
+                        session.user_or_token,
+                        options=[joinedload(DefaultFollowupRequest.allocation)],
+                    )
+                )
+                .unique()
+                .all()
+            )
+
+            default_followup_request_data = []
+            for request in default_followup_requests:
+                default_followup_request_data.append(
+                    {
+                        **request.to_dict(),
+                        'allocation': request.allocation.to_dict(),
+                    }
+                )
+
+            return self.success(data=default_followup_request_data)
+
+    @auth_or_token
+    def delete(self, default_followup_request_id):
+        """
+        ---
+        description: Delete a default follow-up request
+        tags:
+          - filters
+        parameters:
+          - in: path
+            name: default_followup_request_id
+            required: true
+            schema:
+              type: integer
+        responses:
+          200:
+            content:
+              application/json:
+                schema: Success
+        """
+
+        with self.Session() as session:
+
+            stmt = DefaultFollowupRequest.select(session.user_or_token).where(
+                DefaultFollowupRequest.id == default_followup_request_id
+            )
```

Comment:
I think you should also specify that you need permissions to delete when making the select statement.

Reply:
Good point.